src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -179,6 +184,17 @@ private void checkProgrammingExerciseForError(ProgrammingExercise exercise) {
         if (solutionRepositoryUrl != null && !versionControlService.get().repositoryUrlIsValid(solutionRepositoryUrl)) {
             throw new BadRequestAlertException("The Solution Repository URL seems to be invalid.", "Exercise", ErrorKeys.INVALID_SOLUTION_REPOSITORY_URL);
         }
+
+        // if the updated exercise has only automatic feedback at least one test case weight has to be >0
+        // otherwise students can never reach a score >0%
+        if (exercise.getAssessmentType() == AssessmentType.AUTOMATIC) {
```

Comment:
We already test this in the other file. Could you please combine this to reduce code duplication and use one unified approach?

Reply:
:+1: Moved to a single static method `isTestCaseWeightSumValid` in `ProgrammingExerciseTestCaseService`.